util/src/test-support/java/tech/pegasys/artemis/util/EventSink.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.util;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import java.util.ArrayList;
+import java.util.List;
+
+public class EventSink<T> {
+
+  private final List<T> events = new ArrayList<>();
+  private final Class<T> eventType;
+
+  private EventSink(final Class<T> eventType) {
+    this.eventType = eventType;
+  }
+
+  public static <T> List<T> capture(final EventBus eventBus, Class<T> eventType) {
```

Comment:
Why do you take the eventType as an argument to this class? You already take it in the constructor.

Reply:
It's a static method which invokes the constructor and registers the instance with the event bus.